src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestState.cs

```diff
@@ -139,6 +140,16 @@ public void PinReceiveBuffer(byte[] buffer)
             }
         }
 
+        public void DisposeCtrReadFromResponseStream()
+        {
+            if (CtrReadFromResponseStream != null)
+            {
+                WinHttpTraceHelper.Trace("WinHttpRequestState.DisposeCtrReadFromResponseStream: disposing ctr");
+                CtrReadFromResponseStream.Dispose();
+                CtrReadFromResponseStream = default(CancellationTokenRegistration);
```

Comment:
>  I can't do the assignment to null since it is a value type and I got a compiler error.

Then the check `!= null` will always be true and needs to be changed.